lib/_http_client.js

```diff
@@ -80,14 +80,25 @@ function ClientRequest(options, cb) {
           'DeprecationWarning', 'DEP0109');
       }
     }
-  } else if (options && options[searchParamsSymbol] &&
-             options[searchParamsSymbol][searchParamsSymbol]) {
+  } else if (input && input[searchParamsSymbol] &&
+             input[searchParamsSymbol][searchParamsSymbol]) {
     // url.URL instance
-    options = urlToOptions(options);
+    input = urlToOptions(input);
   } else {
-    options = util._extend({}, options);
+    if (cb) throw new ERR_INVALID_ARG_TYPE('url', ['string', 'URL'], input);
```

Comment:
I think this would make this PR a semver-major change, and I'd like to avoid that if possible. How about we just remove this check?

Reply:
done